R/bayesiannetworkanalysis.R

```diff
@@ -0,0 +1,1297 @@
+#
+# Copyright (C) 2018 University of Amsterdam
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+BayesianNetworkAnalysis <- function(jaspResults, dataset, options) {
+  
+  options[["missingValues"]] <- "listwise" # Unfortunately BDgraph does not work with pairwise missing values
+  
+  dataset <- .networkAnalysisReadData(dataset, options)
+
+  mainContainer <- .bayesianNetworkAnalysisSetupMainContainerAndTable(jaspResults, dataset, options)
+  .bayesianNetworkAnalysisErrorCheck(mainContainer, dataset, options)
+  
+  network <- .bayesianNetworkAnalysisRun(mainContainer, dataset, options)
+  
+  .bayesianNetworkAnalysisMainTable        (mainContainer, dataset, options, network)
+  .bayesianNetworkAnalysisWeightMatrixTable(mainContainer, network, options)
+  .bayesianNetworkAnalysisEdgeEvidenceTable(mainContainer, network, options)
+  .bayesianNetworkAnalysisPlotContainer    (mainContainer, network, options)
+  .bayesianNetworkAnalysisCentralityTable  (mainContainer, network, options)
+  
+  return()
+}
+
+.bayesianNetworkAnalysisSetupMainContainerAndTable <- function(jaspResults, dataset, options) {
+  
+  mainContainer <- jaspResults[["mainContainer"]]
+  if (is.null(mainContainer)) {
+    mainContainer <- createJaspContainer(dependencies = c("variables", "groupingVariable", "estimator",
+                                                          "burnin", "iter", "gprior", "dfprior"))
+    jaspResults[["mainContainer"]] <- mainContainer
+  }
+  .bayesianNetworkAnalysisMainTableMeta(mainContainer, dataset, options)
+  
+  return(mainContainer)
+}
+
+.bayesianNetworkAnalysisMainTableMeta <- function(mainContainer, dataset, options) {
+  
+  if (is.null(mainContainer[["generalTable"]])) {
+    
+    tb <- createJaspTable(gettext("Summary of Network"), position = 1, dependencies = c(
+      "minEdgeStrength")) 
```

Comment:
```suggestion
    tb <- createJaspTable(gettext("Summary of Network"), position = 1, dependencies = "minEdgeStrength") 
```
we don't enforce a line width!